<!-- Creator     : groff version 1.22.4 -->
<!-- CreationDate: Mon May 29 22:56:13 2023 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p       { margin-top: 0; margin-bottom: 0; vertical-align: top }
       pre     { margin-top: 0; margin-bottom: 0; vertical-align: top }
       table   { margin-top: 0; margin-bottom: 0; vertical-align: top }
       h1      { text-align: center }
</style>
<title>std::nested_exception</title>

</head>
<body>
<h1>std::nested_exception</h1>



<h2>NAME
<a name="NAME"></a>
</h2>



<p style="margin-left:11%; margin-top: 1em">std::nested_exception</p>

<h2>SYNOPSIS
<a name="SYNOPSIS"></a>
</h2>


<p style="margin-left:11%; margin-top: 1em"><tt>#include
&lt;exception&gt;</tt></p>

<p style="margin-left:11%; margin-top: 1em"><b>Public
Member Functions</b></p>

<p style="margin-left:17%;"><b>nested_exception</b> ()
noexcept <br>
The default constructor stores the current exception (if
any). <b><br>
nested_exception</b> (const <b>nested_exception</b> &amp;)
noexcept=default <b><br>
exception_ptr nested_ptr</b> () const noexcept <br>
Access the stored exception. <b><br>
nested_exception</b> &amp; <b>operator=</b> (const
<b>nested_exception</b> &amp;) noexcept=default <br>
void <b>rethrow_nested</b> () const <br>
Rethrow the stored exception, or terminate if none was
stored.</p>

<h2>Detailed Description
<a name="Detailed Description"></a>
</h2>


<p style="margin-left:11%; margin-top: 1em">Mixin class
that stores the current exception.</p>

<p style="margin-left:11%; margin-top: 1em">This type can
be used via <tt>std::throw_with_nested</tt> to store the
current exception nested within another exception.</p>


<p style="margin-left:11%; margin-top: 1em"><b>Since</b></p>

<p style="margin-left:17%;">C++11</p>

<p style="margin-left:11%; margin-top: 1em"><b>See
also</b></p>

<p style="margin-left:17%;">std::throw_with_nested</p>

<h2>Constructor &amp; Destructor Documentation
<a name="Constructor &amp; Destructor Documentation"></a>
</h2>



<p style="margin-left:11%; margin-top: 1em"><b>std::nested_exception::nested_exception
()</b> <tt>[inline]</tt><b>,</b> <tt>[noexcept]</tt> <br>
The default constructor stores the current exception (if
any).</p>

<h2>Member Function Documentation
<a name="Member Function Documentation"></a>
</h2>



<p style="margin-left:11%; margin-top: 1em"><b>exception_ptr
std::nested_exception::nested_ptr () const</b>
<tt>[inline]</tt><b>,</b> <tt>[noexcept]</tt> <br>
Access the stored exception.</p>

<p style="margin-left:11%; margin-top: 1em"><b>void
std::nested_exception::rethrow_nested () const</b>
<tt>[inline]</tt> <br>
Rethrow the stored exception, or terminate if none was
stored.</p>

<p style="margin-left:11%; margin-top: 1em">References
<b>std::rethrow_exception()</b>, and
<b>std::terminate()</b>.</p>

<h2>Author
<a name="Author"></a>
</h2>


<p style="margin-left:11%; margin-top: 1em">Generated
automatically by Doxygen for libstdc++ from the source
code.</p>
<hr>
</body>
</html>
